### PR TITLE
update evss letters/gi status responses

### PR DESCRIPTION
**Pull request by kreek**
- returns data: null rather than empty response struct for non 200
- added INT data to gi bill status spec and adjusted response
- refactored evss base response

### CODE REVIEW

File: lib/evss/gi_bill_status/gi_bill_status_response.rb [hunk 1]
```diff
@@ -5,19 +5,11 @@
...
-      attribute :va_file_number, String
-      attribute :regional_processing_office, String
-      attribute :eligibility_date, String
-      attribute :delimiting_date, String
-      attribute :percentage_benefit, Integer
-      attribute :original_entitlement, Integer
-      attribute :used_entitlement, Integer
-      attribute :remaining_entitlement, Integer
-      attribute :enrollments, Array[Object]
+      attribute :body, EVSS::GiBillStatus::Post911GIBillStatus
```
Comment (robbiethegeek-usds) [inline lib/evss/gi_bill_status/gi_bill_status_response.rb]: I am a huge proponent of this paradigm.  This looks great.

**Comment by robbiethegeek-usds** [comment]
@kreek this looks great to me @omgitsbillryan please review :shipit:

**Comment by omgitsbillryan** [comment]
This connects:
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3451

File: lib/evss/response.rb [hunk 2]
```diff
@@ -7,10 +7,21 @@ class Response < Common::Base
...
-      self.status = status
-      super(response.body) if response
+    def initialize(response)
+      case response
+      when Faraday::Response
+        self.status = response.status
+        parse_body(response.body)
+      else
+        self.status = response[:status]
+        parse_body(response[:body]) if response[:body]
```
Comment (omgitsbillryan) [inline lib/evss/response.rb]: This looks to be doing too much.  Might I recommend that the method signature look like this:
```ruby
def initialize(status, attributes)
```
and push the responsibility of teasing actual data out of the Faraday response down into the respective `EVSS::Response` inherited response class.  For example, `GiBillStatusResponse` would have:
```ruby
def initialize(status, response = nil)
  attributes = response.body['chapter33_education_info']
  super(status, attributes)
end
```
and `BeneficiaryResponse` would have
```ruby
def initialize(status, response = nil)
  attributes = response.body
  super(status, attributes)
end
```

File: lib/evss/gi_bill_status/gi_bill_status_response.rb (hunk 1 above)
Comment (omgitsbillryan) [inline lib/evss/gi_bill_status/gi_bill_status_response.rb]: I'm don't like that there's an entire object with one attribute called `body`.  I think the `FaradayResponse.body` is just fine and that this abstraction is redundant.

File: lib/evss/gi_bill_status/amendment.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+require 'common/models/base'
+
+module EVSS
+  module GiBillStatus
+    class Amendment < Common::Base
```
Comment (omgitsbillryan) [inline lib/evss/gi_bill_status/amendment.rb]: :+1: Thanks for this!

File: lib/evss/gi_bill_status/enrollment.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+require 'common/models/base'
+
+module EVSS
+  module GiBillStatus
+    class Enrollment < Common::Base
```
Comment (omgitsbillryan) [inline lib/evss/gi_bill_status/enrollment.rb]: :+1:

File: lib/evss/response.rb [hunk 5]
```diff
@@ -7,10 +7,21 @@ class Response < Common::Base
     include Common::Client::ServiceStatus
 
     attribute :status, Integer
+    attribute :body, Object
 
-    def initialize(status, response = nil)
-      self.status = status
-      super(response.body) if response
```
Comment (omgitsbillryan) [inline lib/evss/response.rb]: We don't need this `super`?

**Comment by omgitsbillryan** [comment]
Once this gets merged, mind closing the issue:
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3451

**Pull request merged**